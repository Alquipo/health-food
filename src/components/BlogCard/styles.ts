import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: 40rem;
    height: 45rem;
    width: 100%;
    border-radius: ${theme.border.radius};
    overflow: hidden;
    box-shadow: 0 1.1rem 1.5rem 0 rgba(10, 10, 10, 0.3);
    border: none;
  `}
`

export const WrapperImage = styled.div`
  width: 100%;
  height: 50%;
  position: relative;
`

export const Content = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.small};
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    color: ${theme.colors.secondary};
    font-size: ${theme.font.sizes.xxlarge};

    ${media.between('small', 'medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `};

    ${media.lessThan('small')`
      font-size: ${theme.font.sizes.xlarge};
    `};
  `}
`

export const AuthorWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    margin-top: ${theme.spacings.xsmall};

    .AuthorPhoto {
      border-radius: 50%;
    }

    ${media.lessThan('small')`
    margin-top: ${theme.spacings.small};
    `};
  `}
`

export const AuthorName = styled.span`
  ${({ theme }) => css`
    margin-left: ${theme.spacings.xxsmall};
    font-size: ${theme.font.sizes.xxlarge};

    ${media.lessThan('medium')`
      font-size: ${theme.font.sizes.xlarge};
    `};
  `}
`
