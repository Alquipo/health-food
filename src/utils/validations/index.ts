import * as yup from 'yup'

const nascimentoRegex =
  /^((((19|[2-9]\d)\d{2})-(0[13578]|1[02])-(0[1-9]|[12]\d|3[01]))|(((19|[2-9]\d)\d{2})-(0[13456789]|1[012])-(0[1-9]|[12]\d|30))|(((19|[2-9]\d)\d{2})-02-(0[1-9]|1\d|2[0-8]))|(((1[6-9]|[2-9]\d)(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00))-02-29))$/g

const cpfRegex = /^\d{3}.?\d{3}.?\d{3}-?\d{2}$/

const cepRegex = /^([\d]{2})\.*([\d]{3})-*([\d]{3})/

const estadoRegex =
  /^(AC|AL|AP|AM|BA|CE|DF|ES|GO|MA|MT|MS|MG|PA|PB|PR|PE|PI|RJ|RN|RS|RO|RR|SC|SP|SE|TO)$/i

export const fieldsValidationSchema = yup.object({
  name: yup.string().required('Is not allowed to be empty'),
  birthDate: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(nascimentoRegex, 'must be a valid email'),
  cpf: yup
    .string()
    .max(14, 'Must be 14 characters')
    .required('Is not allowed to be empty')
    .matches(cpfRegex, 'Invalid CPF'),
  zipCode: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(cepRegex, 'Invalid ZipCode'),
  address: yup.string().required('Is not allowed to be empty'),
  district: yup.string().required('Is not allowed to be empty'),
  number: yup.number().required('Is not allowed to be empty'),
  city: yup.string().required('Is not allowed to be empty'),
  state: yup
    .string()
    .required('Is not allowed to be empty')
    .matches(estadoRegex, 'Invalid State')
})
