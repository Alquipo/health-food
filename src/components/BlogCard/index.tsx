import * as S from './styles'
import Image from 'next/image'

export type BlogCardProps = {
  image: string
  title: string
  authorName: string
  authorPhoto: string
}
export default function BlogCard({
  image,
  title,
  authorName,
  authorPhoto
}: BlogCardProps) {
  return (
    <S.Wrapper>
      <S.WrapperImage>
        <Image src={image} alt={title} objectFit="cover" layout="fill" />
      </S.WrapperImage>

      <S.Content>
        <S.Title>{title}</S.Title>

        <S.AuthorWrapper>
          <Image
            src={authorPhoto}
            alt={authorName}
            objectFit="cover"
            layout="fixed"
            width="50"
            height="50"
            className="AuthorPhoto"
          />
          <S.AuthorName>{authorName}</S.AuthorName>
        </S.AuthorWrapper>
      </S.Content>
    </S.Wrapper>
  )
}
