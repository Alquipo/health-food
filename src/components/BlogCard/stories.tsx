import { Story, Meta } from '@storybook/react'
import BlogCard, { BlogCardProps } from '.'

export default {
  title: 'BlogCard',
  component: BlogCard,
  args: {
    image: '/img/bloco_image_1.svg',
    title: 'Quick-start guide to nuts and seeds',
    authorName: 'Kevin Ibrahim',
    authorPhoto: '/img/ibrahim.png'
  }
} as Meta

export const Default: Story<BlogCardProps> = (args) => (
  <div style={{ maxWidth: '34rem' }}>
    <BlogCard {...args} />
  </div>
)
