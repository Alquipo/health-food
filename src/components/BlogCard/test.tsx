import { render, screen } from 'utils/test-utils'

import BlogCard from '.'

const blogCardMock = {
  image: '/img/bloco_image_1.svg',
  title: 'Quick-start guide to nuts and seeds',
  authorName: 'Kevin Ibrahim',
  authorPhoto: '/img/ibrahim.png'
}

describe('<BlogCard/>', () => {
  it('should render a BlogCard Section correctly', () => {
    const { container } = render(<BlogCard {...blogCardMock} />)

    expect(
      screen.getByRole('img', { name: blogCardMock.title })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('img', { name: blogCardMock.authorName })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: blogCardMock.title })
    ).toBeInTheDocument()

    expect(screen.getByText(blogCardMock.authorName)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
