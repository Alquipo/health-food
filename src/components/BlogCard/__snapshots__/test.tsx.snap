// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogCard/> should render a BlogCard Section correctly 1`] = `
.c0 {
  max-width: 40rem;
  height: 45rem;
  width: 100%;
  border-radius: 0.6rem;
  overflow: hidden;
  box-shadow: 0 1.1rem 1.5rem 0 rgba(10,10,10,0.3);
  border: none;
}

.c1 {
  width: 100%;
  height: 50%;
  position: relative;
}

.c2 {
  padding: 2.4rem;
}

.c3 {
  color: #1D164D;
  font-size: 2.8rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1.6rem;
}

.c4 .AuthorPhoto {
  border-radius: 50%;
}

.c5 {
  margin-left: 0.8rem;
  font-size: 2.8rem;
}

@media (min-width:450px) and (max-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (max-width:450px) {
  .c3 {
    font-size: 2.0rem;
  }
}

@media (max-width:450px) {
  .c4 {
    margin-top: 2.4rem;
  }
}

@media (max-width:768px) {
  .c5 {
    font-size: 2.0rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <noscript />
      <img
        alt="Quick-start guide to nuts and seeds"
        data-nimg="true"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
      />
    </div>
  </div>
  <div
    class="c2"
  >
    <h1
      class="c3"
    >
      Quick-start guide to nuts and seeds
    </h1>
    <div
      class="c4"
    >
      <div
        style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 50px; height: 50px;"
      >
        <noscript />
        <img
          alt="Kevin Ibrahim"
          class="AuthorPhoto"
          data-nimg="true"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </div>
      <span
        class="c5"
      >
        Kevin Ibrahim
      </span>
    </div>
  </div>
</div>
`;
